src/runtime/cpp/lean_runtime.cpp

```diff
@@ -94,6 +100,7 @@ obj_cell::obj_cell(obj_cell const & src, obj const & a1):
     copy_fields(src);
     void ** f = field_addr();
     new (f + src.m_size) obj(a1);
+    *fn_ptr_addr() = src.fn_ptr();
```

Comment:
I believe this was a bug where copying closures would lose the fn_ptr, causing a segfault when using `apply(..)`.